cpp/src/context.cpp

```diff
@@ -72,18 +72,18 @@ std::shared_ptr<Context> createContext(const ConfigMap ucxConfig, const uint64_t
 
 Context::~Context()
 {
-  if (this->_handle != nullptr) ucp_cleanup(this->_handle);
-  ucxx_trace("Context destroyed: %p", this->_handle);
+  if (_handle != nullptr) ucp_cleanup(_handle);
+  ucxx_trace("Context destroyed: %p, UCP handle: %p", this, _handle);
```

Comment:
<!-- Polish comments are like a suggestion, where there is nothing necessarily wrong with the relevant content, there's just some ways to immediately improve the quality.  -->
**polish:** ‏While we're here:

```suggestion
  if (_handle != nullptr) ucp_cleanup(_handle);
  ucxx_trace("Context destroyed: %p, UCP handle: %p", this, _handle);
  _handle = nullptr;
```

And similarly in the other destructors.

Or does this not matter because the object is about to disappear anyway, so we can't access `_handle` after `ucp_cleanup` is called.

Reply:
Setting it to `nullptr` is unnecessary, I'd even argue bad practice (why not then reset/clear _every_ member?). It is undefined behavior to access `_handle` or any other non-shared member during/after destructor runs.